core/src/main/java/org/bitcoinj/wallet/BasicKeyChain.java

```diff
@@ -283,14 +284,21 @@ public boolean removeKey(ECKey key) {
         }
     }
 
+    /**
+     * Returns the earliest creation time of keys in this chain.
+     * @return earliest creation times of keys in this chain,
+     *         {@link Instant#EPOCH} if at least one time is unknown,
+     *         {@link Instant#MAX} if no keys in this chain
+     */
     @Override
-    public long getEarliestKeyCreationTime() {
+    public Instant getEarliestKeyCreationTimeInstant() {
         lock.lock();
         try {
             return hashToKeys.values().stream()
                     .mapToLong(ECKey::getCreationTimeSeconds)
-                    .min()
-                    .orElse(Long.MAX_VALUE);
+                    .mapToObj(t -> t == Long.MAX_VALUE ? Instant.MAX : Instant.ofEpochSecond(t))
+                    .min((t1, t2) -> t1.compareTo(t2))
```

Comment:
Can a method reference be used here?  (Maybe with a `Comparators` helper?)

Reply:
I assume it could. However I think this line of code reads well, without the indirection of a method reference. Your milage may very, but you're probably well ahead in grokking the functional style.

Reply:
Ok, I'm using a static method now. Can you review 108b0d7e3eaf42c8bb87d08ca22f4002e698ae7e if that will work as expected?

I'm stoked that such method does not exist in `Objects`. They is one for non-natural orders, but none for the natural order.

Reply:
Ok, I've reverted the previous attempt and created 7ffa950cefdd28d2cf6a35bece7ef5d64c1ce9c9.